### ISSUE TITLE
 MetricsTracer Interface: Private “state” Field in UpdatedBlackHoleFilterState Method

### ISSUE THREAD

**Issue by cpeliciari** [libp2p/go-libp2p#2890]
Hi,

I am working on monitoring a private system that uses libp2p. Recently, I encountered an issue with the MetricsTracer interface located in p2p/net/swarm/swarm_metrics.go.

The MetricsTracer interface has a method UpdatedBlackHoleFilterState that includes a state field. However, this field is private, which prevents the recreation of the MetricsTracer to collect connection and dial data.

```
type MetricsTracer interface {
	OpenedConnection(network.Direction, crypto.PubKey, network.ConnectionState, ma.Multiaddr)
	ClosedConnection(network.Direction, time.Duration, network.ConnectionState, ma.Multiaddr)
	CompletedHandshake(time.Duration, network.ConnectionState, ma.Multiaddr)
	FailedDialing(ma.Multiaddr, error, error)
	DialCompleted(success bool, totalDials int)
	DialRankingDelay(d time.Duration)
	UpdatedBlackHoleFilterState(name string, state blackHoleState, nextProbeAfter int, successFraction float64)
}
```

I would like to know if there is a specific reason for the state field to be private.

Thank you in advance for your attention to this matter.

**Comment by sukunrt** [comment]
I think we just missed this. Can you open a PR with a method for getting State?

**Comment by sukunrt** [comment]
Closed by #2917
